contracts/PrizeTierHistoryV2.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.6;
+import "@pooltogether/owner-manager-contracts/contracts/Manageable.sol";
+import "./abstract/IdBinarySearch.sol";
+
+contract PrizeTierHistoryV2 is IdBinarySearch, Manageable {
+
+    struct PrizeTier {
+        uint8 bitRangeSize;
+        uint32 drawId;
+        uint32 maxPicksPerUser;
+        uint32 expiryDuration;
+        uint32 endTimestampOffset;
+        uint256 prize;
+        uint32[16] tiers;
+    }
+
+    uint32[] internal history;
+    mapping(uint32 => PrizeTier) internal prizeTiers;
+
+    /**
+     * @notice Emit when new PrizeTier is added to history
+     * @param drawId    Draw ID
+     * @param prizeTier PrizeTier parameters
+     */
+    event PrizeTierPushed(uint32 indexed drawId, PrizeTier prizeTier);
+
+    /**
+     * @notice Emit when existing PrizeTier is updated in history
+     * @param drawId    Draw ID
+     * @param prizeTier PrizeTier parameters
+     */
+    event PrizeTierSet(uint32 indexed drawId, PrizeTier prizeTier);
+
+    constructor(address owner, PrizeTier[] memory _history) Ownable(owner) {
+        if (_history.length > 0) {
+            inject(_history);
+        }
+    }
+
+    // @inheritdoc DrawIDBinarySearch
+    function getNewestIndex() internal view override returns (uint32) {
+        return uint32(history.length - 1);
+    }
+
+    // @inheritdoc DrawIDBinarySearch
+    function getIdForIndex(uint256 index) internal view override returns (uint32) {
+        return history[index];
+    }
+
+    function getPrizeTier(uint32 drawId) external view returns (PrizeTier memory) {
+        require(drawId > 0, "PrizeTierHistoryV2/draw-id-not-zero");
+        return prizeTiers[history[_binarySearch(drawId)]];
+    }
+
+    function getPrizeTierList(uint32[] calldata _drawIds) external view returns (PrizeTier[] memory) {
+        PrizeTier[] memory _data = new PrizeTier[](_drawIds.length);
```

Comment:
You can store `_drawIds.length` in a variable to save a bit on gas in the for loop.